client/src/components/FormField/RegisteredMFAMethodListField/RegisteredMFAMethodListField.js

```diff
@@ -1,129 +1,429 @@
 import PropTypes from 'prop-types';
 import React, { Component } from 'react';
-import { Button, Modal, ModalHeader, ModalBody } from 'reactstrap';
+import { Button, Modal, ModalBody, ModalHeader } from 'reactstrap';
 import { inject } from 'lib/Injector'; // eslint-disable-line
 import { compose } from 'redux';
 import { connect } from 'react-redux';
 import classnames from 'classnames';
 import methodShape from 'types/registeredMethod';
-
 import AccountResetUI from './AccountResetUI';
 import MethodListItem from './MethodListItem';
-import { showScreen, chooseMethod, setAvailableMethods } from 'state/mfaRegister/actions';
-import { SCREEN_CHOOSE_METHOD } from 'components/Register';
+import {
+  addAvailableMethod,
+  chooseMethod,
+  setAvailableMethods,
+  showScreen,
+} from 'state/mfaRegister/actions';
+import {
+  SCREEN_CHOOSE_METHOD,
+  SCREEN_REGISTER_METHOD,
+  SCREEN_INTRODUCTION
+} from 'components/Register';
 import Title from '../../Register/Title';
+import Config from 'lib/Config'; // eslint-disable-line
+import confirm from '@silverstripe/reactstrap-confirm';
 
 const fallbacks = require('../../../../lang/src/en.json');
 
 class RegisteredMFAMethodListField extends Component {
   constructor(props) {
     super(props);
 
+    const { initialAvailableMethods, registeredMethods } = props;
```

Comment:
Adding 444 lines to a component seems wrong to me. Any way you could split it up?

Reply:
Ok I've done a bit of splitting up now. Let me know if that suffices.